engine/src/test/resources/testAppPackage/mydtapp/src/main/java/com/example/mydtapp/TestModule.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.example.mydtapp;
+
+import org.apache.hadoop.conf.Configuration;
+import com.esotericsoftware.kryo.NotNull;
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.Module;
+
+/**
+ * This is a simple operator that emits random number
+ */
+
+public class TestModule implements Module
+{
+  @NotNull
+  private String testInput;
+
+  public final transient ProxyOutputPort<String> outputs = new ProxyOutputPort<String>();
+
+  public final transient ProxyInputPort<String> inputs = new ProxyInputPort<String>();
+
+  @Override
+  public void populateDAG(DAG dag, Configuration configuration)
+  {
+    testInput.length();
```

Comment:
This throws NullPointerException if testInput is not properly initialized. Fix provided in this pull request indirectly ensures properties mentioned in properties.xml are initialized correctly.